### PR TITLE
Logs join, and shares and devices tables.

### PR DESCRIPTION
**Pull request by dkorolev**
1. United two logs: Storage mutations and CTFO events.
2. Added two "Shares" tables.
3. Added the "Devices" table.

### CODE REVIEW

File: 2016_08_01_conversion/convert.cc [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+  void operator()(const current::midichlorians::ios::iOSDeviceInfo& input_device_info) {
+    typename CTFO_2016_08_01::CTFOEvent ctfo_event;
+    ctfo_event = typename CTFO_2016_08_01::TopLevelTransaction();
+    auto& output_event = Value<typename CTFO_2016_08_01::TopLevelTransaction>(ctfo_event);
+
+    output_event.meta.begin_us = us + std::chrono::microseconds(1);
+    output_event.meta.end_us = us + std::chrono::microseconds(3);
+
+    typename CTFO_2016_08_01::iOSDeviceInfo output_device_info;
```
Comment (mzhurovich) [inline 2016_08_01_conversion/convert.cc]: Unused?

File: 2016_08_01_conversion/convert.cc [hunk 2]
```diff
@@ -78,7 +78,7 @@ struct CTFOEventToStreamImporter {
 
 #define IMPORT_EVENT(EVENT_TYPE_NAME)                                                                \
   void operator()(const midichlorians_events_to_import::EVENT_TYPE_NAME& input_event) {              \
-    typename CTFO_2016_08_01::CTFOEvent ctfo_event;                                                  \
+    typename CTFO_2016_08_01::CTFOLogEntry ctfo_event;                                               \
```
Comment (mzhurovich) [inline 2016_08_01_conversion/convert.cc]: s/ctfo_event/ctfo_log_entry/?

Comment (mzhurovich) [inline 2016_08_01_conversion/convert.cc]: ctfo_event_log_entry

File: ctfo_server.h [hunk 3]
```diff
@@ -1511,11 +1511,11 @@ class CTFOServer final {
   void UpdateStateOnEvent(const current::midichlorians::ios::iOSGenericEvent& ge) {
     const auto now = current::time::Now();
     try {
-      if (!valid_responses_.count(ge.event)) {
+      if (!supported_events_.count(ge.event)) {
         DebugPrint(Printf("[UpdateStateOnEvent] Unhandled event: '%s'", ge.event.c_str()));
         return;
       }
-      const LOG_EVENT response = valid_responses_.at(ge.event);
+      const LOG_EVENT response = supported_events_.at(ge.event);
```
Comment (mzhurovich) [inline ctfo_server.h]: s/response/event/?

**Comment by mzhurovich** [comment]
👍 👍 👍 💯

File: ctfo_server.h [hunk 4]
```diff
@@ -1757,10 +1782,36 @@ class CTFOServer final {
                   } else {
                     DebugPrint("[UpdateStateOnEvent] Ignoring `ONE_SIGNAL_USER_ID`, no `used_id` provided.\n");
                   }
+                } else if (valid_share_statuses_.count(event)) {
+                  const SHARE_STATUS share_status = valid_share_statuses_.at(event);
+                  DebugPrint(Printf(
+                      "[UpdateStateOnEvent] Share status `%s` for uid='%s', cid='%s',token='%s'",
```
Comment (mzhurovich) [inline ctfo_server.h]: Bug

**Comment by dkorolev** [comment]
СС @grixa : Тhis is live as of half an hour ago. Подробности -- голосом!

File: 2016_08_01_conversion/convert.cc (hunk 2 above)
Comment (dkorolev) [inline 2016_08_01_conversion/convert.cc]: :+1: b49fc65

**Pull request merged**